src/typechecker.rs

```diff
@@ -4766,6 +4766,14 @@ pub fn typecheck_expression(
                 }
             }
 
+            let parent_throws = project.get_scope(scope_id).throws;
+            if fill_size_expr.is_some() && !parent_throws {
+                error = error.or(Some(JaktError::TypecheckError(
+                    "Prefilling an array may throw and needs to be in a try statement or a function marked as throws".to_string(),
+                    *span,
```

Comment:
can we use the span of the `fill_size_expr` as the location of the error?

Reply:
Sure, I changed it :)